test/integration_tests.jl

```diff
@@ -32,25 +32,6 @@ end
     @test (@ballocated avoid_allocations($record, f_matmul_noprealloc)) <= 1520
 end
 
-@testset "check thread-safe" begin
-    f(x, y) = x * y
-    n = Threads.nthreads()
-    results = Vector{Any}(undef, n)
-    As = [rand(4, 4) for _ in 1:n]
-    Bs = [rand(4, 4) for _ in 1:n]
-    _, pf = preallocate(f, As[1], Bs[1])
-    Threads.@threads for k in 1:n
-        results[k] = pf(As[k], Bs[k])
-    end
-
-    # if it's not thread-safe, the result
-    # will be modified by other multiplications
-    # since they will share the same memory
-    for k in 1:n
-        @test results[k] ≈ f(As[k], Bs[k])
-    end
-end
-
```

Comment:
I don't understand how this test ever worked, preallocate is _not_ per-thread so this test _should_ have been failing?

Reply:
`preallocate` isn't per thread.
But invoking a `pf` *is* per thread.
https://github.com/oxinabox/AutoPreallocation.jl/blob/1013ac618749ac3bd3534162cb6c176e4c6d19bf/src/preallocate.jl#L32-L39
This should pass

Reply:
Ah, I missed that, thanks. It was failing for me locally on 1.5, I'll take a look.

Reply:
I think the `copy` was just on the wrong branch, test added back now.